### ISSUE TITLE
Relative path in configuration.yaml files

### ISSUE THREAD

**Issue by chrismbirmingham** [interaction-lab/HARMONI#17]
Should be using a relative path if possible (maybe something like $(find harmoni_pattern) to get the base path?), though this may depend on how this file is read in.
e.g. in harmoni_core/harmoni_pattern/config/configuration.yaml
pattern_scripting: /root/harmoni_catkin_ws/src/HARMONI/harmoni_pattern/pattern_scripting/dialogue.json

**Comment by RMichaelSwan** [comment]
There was a note during today's meeting about how people were committing back and forth small changes to some base configuration file (relating to the file mentioned in this issue). We want the file to be committed as some base version, but we don't want our unique version to necessarily be included in commits that are made. To do this for files which are tracked by git, .gitignore will not work. Instead, you have to make git assume that the indexed file has not changed, thus ignoring any changes you make to the file until you undo the setting.

In order ignore changed to a tracked file: `git update-index --assume-unchanged <file>`
In order to undo it: `git update-index --no-assume-unchanged <file>`
To get a list of files that have this option set: `git ls-files -v|grep '^h' `

_Note: One thing that needs testing is whether this setting gets pushed to the repo for everyone or only works locally._

Source: https://stackoverflow.com/questions/17195861/undo-git-update-index-assume-unchanged-file

**Comment by RMichaelSwan** [comment]
Fixed in recent commit: a0d9664c236e37bcabe8ed71e970c5eddb190c40
This was done by enabling parsing of substitution args in relevant launch/test files in the offending yaml files and then inserting relative/env args inplace. Note use of `subst_value`.

launch file: 
```xml
<rosparam file="$(find harmoni_pattern)/config/configuration.yaml" subst_value="True"/>
```
config yaml: 
```yaml
credential_path: "$(env HOME)/.gcp/private-keys.json"
```
